exoplayer-codelab-01/src/main/java/com/example/exoplayer/PlayerActivity.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2017 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.exoplayer
+
+import android.annotation.SuppressLint
+import android.os.Bundle
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import com.example.exoplayer.databinding.ActivityPlayerBinding
+import com.google.android.exoplayer2.MediaItem
+import com.google.android.exoplayer2.SimpleExoPlayer
+import com.google.android.exoplayer2.util.Util
+
+/**
+ * A fullscreen activity to play audio or video streams.
+ */
+class PlayerActivity : AppCompatActivity() {
+
+    private val viewBinding by lazy { ActivityPlayerBinding.inflate(layoutInflater) }
+
+    private var player: SimpleExoPlayer? = null
+
+    private var playWhenReady = true
+    private var currentWindow = 0
+    private var playbackPosition: Long = 0
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(viewBinding.root)
+    }
+
+    public override fun onStart() {
+        super.onStart()
+        if (Util.SDK_INT > 23) {
+            initializePlayer()
+        }
+    }
+
+    public override fun onResume() {
+        super.onResume()
+        hideSystemUi()
+        if (Util.SDK_INT <= 23 || player == null) {
+            initializePlayer()
+        }
+    }
+
+    public override fun onPause() {
+        super.onPause()
+        if (Util.SDK_INT <= 23) {
+            releasePlayer()
+        }
+    }
+
+    public override fun onStop() {
+        super.onStop()
+        if (Util.SDK_INT > 23) {
+            releasePlayer()
+        }
+    }
+
+    private fun initializePlayer() {
+        player = SimpleExoPlayer.Builder(this)
+            .build()
+            .also {
```

Comment:
Opted for `it` based on conversations here to be more idiomatic: https://github.com/googlecodelabs/exoplayer-intro/pull/62 as the more explicit names can be a bit noisy. It could also just be that I didn't choose very good explicit names lmao. What would you suggest naming `it` to?

Reply:
Some alternatives: 

1. Use `apply` instead of `also`. This would avoid using `it`. 
2. Rename `it` to `exoPlayer` since `it` is the ExoPlayer implementation of the `Player` interface`.

Reply:
Opted for `apply { -> exoPlayer }` as the semantics of `this` got hard to discern because of the `playWhenReady` `var` .

It would have required:

```kotlin
apply {
    playWhenReady = this@PlayerActivity.playWhenReady
}
```